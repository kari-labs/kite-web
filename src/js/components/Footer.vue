<template>
    <v-footer app absolute inset color="primary" height="auto">
        <v-card flat tile color="primary" class="text-xs-center">
            <v-card-text>
                <v-btn v-for="icon in icons" :key="icon" :aria-label="icon" class="mx-3 secondary-text" icon>
                    <v-icon size="24px" color="secondary">{{icon}}</v-icon>
                </v-btn>
            </v-card-text>
            <v-card-text class="pt-0 secondary-text product-sans">
                Use of this system is limited to authorized persons of New England Institute of Technology.
            </v-card-text>
            <v-divider></v-divider>
            <v-card-text class="secondary-text product-sans">
                &copy;2018 — <strong>Joseph Zurowski</strong> <br />
                Product Sans &copy;2015 — 2018 <strong>google_logo</strong>
            </v-card-text>
        </v-card>
    </v-footer>
</template>

<script>
    export default {
        data: () => ({
            icons: [
                'fab fa-facebook',
                'fab fa-twitter',
                'fab fa-google-plus',
                'fab fa-instagram'
            ]
        })
    }
</script>