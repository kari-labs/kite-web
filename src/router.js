import Vue from 'vue'
import Router from 'vue-router'

Vue.use(Router)

export default new Router({
  mode: 'history',
  base: '/',
  routes: [
    {
        path: '/',
        name: 'listContainers',
        component: () => import(/* webpackChunkName: "containerDashboard" */ './views/ContainerDashboard.vue')
    }
  ]
})
