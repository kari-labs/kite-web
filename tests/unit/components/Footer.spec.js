import { mount } from '@vue/test-utils'
import Vue from 'vue'
import Vuetify from 'vuetify'
import router from '@/router'
import Footer from '@/components/Footer.vue'

describe('Footer', () => {
    let wrapper

    beforeEach(() => {
        Vue.use(router)
        Vue.use(Vuetify)

        wrapper = mount(Footer, { router })
    })

    it('should render Vuetify components', () => {
        expect(wrapper.contains('footer.v-footer')).toBe(true)
    })

    it('should render social media links correctly', () => {
        wrapper.vm.icons = [
            'fab fa-twitch',
            'fab fa-google-plus',
            'fab fa-steam'
        ]

        expect(wrapper.element).toMatchSnapshot()
    })
})