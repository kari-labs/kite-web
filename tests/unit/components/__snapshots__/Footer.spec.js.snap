// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render social media links correctly 1`] = `
<footer
  class="v-footer v-footer--absolute v-footer--inset theme--light primary"
  style="height: auto;"
>
  <div
    class="text-xs-center v-card v-card--flat v-card--tile theme--light primary"
  >
    <div
      class="v-card__text"
    >
      <button
        aria-label="fab fa-twitch"
        class="mx-3 secondary-text v-btn v-btn--icon theme--light"
        type="button"
      >
        <div
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon fab fa-twitch theme--light secondary--text"
            style="font-size: 24px;"
          />
        </div>
      </button>
      <button
        aria-label="fab fa-google-plus"
        class="mx-3 secondary-text v-btn v-btn--icon theme--light"
        type="button"
      >
        <div
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon fab fa-google-plus theme--light secondary--text"
            style="font-size: 24px;"
          />
        </div>
      </button>
      <button
        aria-label="fab fa-steam"
        class="mx-3 secondary-text v-btn v-btn--icon theme--light"
        type="button"
      >
        <div
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon fab fa-steam theme--light secondary--text"
            style="font-size: 24px;"
          />
        </div>
      </button>
    </div>
     
    <div
      class="v-card__text pt-0 secondary-text"
    >
      
            Use of this system is limited to authorized persons of New England Institute of Technology.
        
    </div>
     
    <hr
      class="v-divider theme--light"
    />
     
    <div
      class="v-card__text secondary-text"
    >
      
            ©2018 — 
      <strong>
        Joseph Zurowski
      </strong>
       
      <br />
    </div>
  </div>
</footer>
`;
